File: src/main/java/com/dh/ondot/member/infra/dto/ApplePublicKeyResponse.java
package com.dh.ondot.member.infra.dto;

import java.util.List;
import java.util.Optional;

public record ApplePublicKeyResponse(
        List<Key> keys
) {
    public record Key(
            String kty,
            String kid,
            String use,
            String alg,
            String n,
            String e
    ){}

    public Optional<Key> getMatchedKey(String kid, String alg) {
        return keys.stream()
                .filter(k -> k.kid().equals(kid) && k.alg().equals(alg))
                .findFirst();
    }
}

File: src/main/kotlin/com/dh/ondot/member/infra/dto/ApplePublicKeyResponse.kt
package com.dh.ondot.member.infra.dto

data class ApplePublicKeyResponse(
    val keys: List<Key>,
) {
    data class Key(
        val kty: String,
        val kid: String,
        val use: String,
        val alg: String,
        val n: String,
        val e: String,
    )

    fun getMatchedKey(kid: String, alg: String): Key? {
        return keys.firstOrNull { it.kid == kid && it.alg == alg }
    }
}
